sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/client/traits/AmqpTrait.java

```diff
@@ -53,9 +53,9 @@
     /**
      * Allows for setting common properties such as application ID, headers, etc.
      *
-     * @param clientOptions A configured instance of {@link ClientOptions}.
+     * @param clientOptions A configured instance of {@link com.azure.core.amqp.AmqpClientOptions}.
      * @return Returns the same concrete type with the appropriate properties updated, to allow for fluent chaining of
      *      operations.
      */
-    T clientOptions(ClientOptions clientOptions);
+    T clientOptions(AmqpClientOptions clientOptions);
```

Comment:
I think this should be an overload, not a replacement?

Reply:
OK, I got it.

Reply:
Paging @kasobol-msft and @srnagar to weigh in here. On the `HttpTrait` we do not have an overload here, and I'm not convinced we should have this here either. We 'solved' the problem on the HTTP side with javadoc: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/client/traits/HttpTrait.java#L122

Reply:
We should be consistent with Http services here.

Reply:
Got it. I'll add a new method `amqpClientOptions(AmqpClientOptions amqpClientOptions)` and keep the `clientOptions(ClientOptions clientOptions)`.